server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -204,26 +235,36 @@ public ClusterState execute(ClusterState currentState) {
                     throw new InvalidSnapshotNameException(
                             repository.getMetadata().name(), snapshotName, "snapshot with the same name already exists");
                 }
+                final SnapshotsInProgress snapshots = currentState.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY);
+                final List<SnapshotsInProgress.Entry> runningSnapshots = snapshots.entries();
+                if (runningSnapshots.stream().anyMatch(s -> {
+                    final Snapshot running = s.snapshot();
+                    return running.getRepository().equals(repositoryName) && running.getSnapshotId().getName().equals(snapshotName);
+                })) {
+                    throw new InvalidSnapshotNameException(
+                            repository.getMetadata().name(), snapshotName, "snapshot with the same name is already in-progress");
+                }
                 validate(repositoryName, snapshotName, currentState);
+                final boolean concurrentOperationsAllowed = currentState.nodes().getMinNodeVersion().onOrAfter(FULL_CONCURRENCY_VERSION);
                 final SnapshotDeletionsInProgress deletionsInProgress =
-                    currentState.custom(SnapshotDeletionsInProgress.TYPE, SnapshotDeletionsInProgress.EMPTY);
-                if (deletionsInProgress.hasDeletionsInProgress()) {
+                        currentState.custom(SnapshotDeletionsInProgress.TYPE, SnapshotDeletionsInProgress.EMPTY);
+                if (deletionsInProgress.hasDeletionsInProgress() && concurrentOperationsAllowed == false) {
                     throw new ConcurrentSnapshotExecutionException(repositoryName, snapshotName,
                         "cannot snapshot while a snapshot deletion is in-progress in [" + deletionsInProgress + "]");
```

Comment:
When backporting, we could maybe indicate in the error message that concurrent snapshot/deletions are possible in version 7.9?

Reply:
:+1: Right, I put down a note for that when doing the back-port work.